cpp/src/arrow/CMakeLists.txt

```diff
@@ -24,6 +24,8 @@ install(FILES
   schema.h
   table.h
   type.h
+  type_fwd.h
```

Comment:
The Kudu conventions seem fine to me. As long as we have some kind of convention I'm happy with it.